src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/SqlCommand/SqlCommandCancelTest.cs

```diff
@@ -13,30 +13,59 @@ namespace Microsoft.Data.SqlClient.ManualTesting.Tests
     public static class SqlCommandCancelTest
     {
         // Shrink the packet size - this should make timeouts more likely
-        private static readonly string s_connStr = (new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) { PacketSize = 512 }).ConnectionString;
+        private static readonly string tcp_connStr = (new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) { PacketSize = 512 }).ConnectionString;
+        private static readonly string np_connStr = (new SqlConnectionStringBuilder(DataTestUtility.NPConnectionString) { PacketSize = 512 }).ConnectionString;
 
         [CheckConnStrSetupFact]
         public static void PlainCancelTest()
         {
-            PlainCancel(s_connStr);
+            PlainCancel(tcp_connStr);
+        }
+
+        [CheckConnStrSetupFact]
```

Comment:
With the TCP/NP separation, you'll need to change CheckConnStrSetupFact to something else to check if the NP connection string is configured.